ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -99,7 +107,9 @@
   private Offset prevSafeEndPoint = null;
   // reset key refers to the first StoreKey that is added to the index segment
   private Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey = null;
-  protected ConcurrentSkipListMap<StoreKey, IndexValue> index = null;
+  private ConcurrentSkipListMap<StoreKey, ConcurrentSkipListSet<IndexValue>> index = null;
+
+  static final Comparator<IndexValue> INDEX_VALUE_COMPARATOR = Comparator.comparing(IndexValue::getOffset);
```

Comment:
I was thinking about this too - I was just wondering if there is anyone who relies on a non offset order but since `IndexValue` doesn't have a `compareTo()` method, that's probably not happening. `IndexValue` logically doesn't have any other field to sort on anyway. I'll take this suggestion and implement the `Comparable` interface. Less error prone for future changes too.